### PR TITLE
feat(modal): remove replace usage

### PR DESCRIPTION
**Pull request by wesleycho**
- Remove `replace: true` usage

BREAKING CHANGE: This removes `replace: true` usage, causing some structural changes to the HTML - the major part here is that there is no more backdrop template, and the top level elements for the window/backdrop elements lose a little flexibility. See documentation examples for new structure.

### CODE REVIEW

**Comment by TechCraver1** [comment]
@wesleycho Milestone 2.0.1 for this.

**Comment by wesleycho** [comment]
? This is going to be in the upcoming 2.0.0 release.

File: src/modal/modal.js [hunk 1]
```diff
@@ -493,13 +497,22 @@ angular.module('ui.bootstrap.modal', ['ui.bootstrap.stackedMap', 'ui.bootstrap.p
         topModalIndex = previousTopOpenedModal ? parseInt(previousTopOpenedModal.value.modalDomEl.attr('index'), 10) + 1 : 0;
         var angularDomEl = angular.element('<div uib-modal-window="modal-window"></div>');
         angularDomEl.attr({
+          'class': 'modal',
           'template-url': modal.windowTemplateUrl,
-          'window-class': modal.windowClass,
```
Comment (icfantv) [inline src/modal/modal.js]: do we need to document that this attribute is removed as part of the breaking changes?

Comment (wesleycho) [inline src/modal/modal.js]: The class is meant for adding to element the class, but that is now happening without needing a binding.

**Pull request closed without merging**